sentinelsat/sentinel.py

```diff
@@ -802,7 +802,11 @@ def download_all(
 
         failed_prods = {}
         for pid, info in product_infos.items():
-            if pid not in downloaded_prods and pid not in retrieval_scheduled:
+            if (
+                pid not in downloaded_prods
+                and pid not in retrieval_scheduled
+                and not dl_tasks[pid].cancelled()
+            ):
                 info["exception"] = dl_tasks[pid].exception()
                 failed_prods[pid] = info
```

Comment:
The cancelled products should still be marked as failed.

```suggestion
            if pid not in downloaded_prods and pid not in retrieval_scheduled:
                if not dl_tasks[pid].cancelled():
                    info["exception"] = dl_tasks[pid].exception()
                failed_prods[pid] = info
```